src/main/java/com/soboleoko/medicalclinic/controller/DoctorController.java

```diff
@@ -0,0 +1,43 @@
+package com.soboleoko.medicalclinic.controller;
+
+import com.soboleoko.medicalclinic.mapper.MedicalClinicMapper;
+import com.soboleoko.medicalclinic.model.Doctor;
+import com.soboleoko.medicalclinic.model.DoctorDTO;
+import com.soboleoko.medicalclinic.service.DoctorService;
+import lombok.RequiredArgsConstructor;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+@RestController
+@RequiredArgsConstructor
+
+public class DoctorController {
+    private final MedicalClinicMapper mapper;
+    private final DoctorService doctorService;
+
+    @GetMapping("/doctors")
+    public List<DoctorDTO> getDoctors() {
+        return mapper.mapToDoctorDTOList(doctorService.getDoctors());
+    }
+
+    @GetMapping("/doctors/{email}")
+    public DoctorDTO getDoctorByEmail(@PathVariable String email) {
+        return mapper.mapToDoctorDTO(doctorService.getDoctorByEmail(email));
+    }
+
+    @PostMapping("/doctors")
+    public DoctorDTO addDoctor(@RequestBody Doctor doctor) {
+        return mapper.mapToDoctorDTO(doctorService.addDoctor(doctor));
+    }
+
+    @DeleteMapping("/doctors/{email}")
+    public void deleteDoctor(@PathVariable String email) {
+        doctorService.deleteDoctor(email);
+    }
+
+    @PutMapping("/doctors/{email}")
+    public DoctorDTO updateDoctor(@PathVariable String email, @RequestBody Doctor newDoctorData) {
```

Comment:
Tak samo nie ma sprawdzenia na polach? @Valid?

Reply:
Zweryfikuj działanie walidacji

Reply:
Nie uzywaj w controllerach obiektów służacych do perysstencji (przechowywania, zapisu w naszym przypadku w liscie). Dlaczego? Dzięki temu jak bedziemy chcieli wprowadzić zmianę do klasy doktor -> zapisac dodatkowe pole np. modifiedAt czyli kiedy byla modyfikacja na tym doktorze to nie udostepniamy takiej mozliwosci od razu uzytkownikowi. Model (Dto) odpowiedzialny za komunikacje ustala komnunikacje z użytkownikiema a model odpowiedzialny za persystencje ustala co przechowujemy w bazie danych. Dzięki temu zmiana po stronie bazy danych nie wplywa na komunikacje z uzytkownikiem i zmiana po stronie komunikacji z uzytkownikiem nie wplywa na baze danych.